mozci/push.py

```diff
@@ -27,6 +29,31 @@
 MAX_DEPTH = 14
 
 
+def get_hgmo(branch, rev):
+    url = HGMO_AUTOMATION_RELEVANCE_URL.format(branch=branch, rev=rev)
+    r = requests.get(url)
+    r.raise_for_status()
+    return r.json()["changesets"][0]
+
+
+backouts = set()
+
+
+@memoize
+def is_backout(branch, rev):
+    global backouts
+
+    if rev not in backouts:
+        return True
+
+    res = get_hgmo(branch, rev)
+    if len(res["backsoutnodes"]) > 0:
+        backouts.add(rev)
+        return True
+
+    return False
```

Comment:
I think these should be methods on the `Push` class, and then `get_candidate_regressions` should instantiate a new `Push` instance and check `push.backsoutnodes`. Since everything is lazily loaded, there shouldn't be much additional perf impact.

You can set this up as a `memoized_property` the same way we do for `self._hgmo`.

Reply:
The only problem is that a push might contain multiple backouts, we'd have to load json-automationrelevance for all commits of a push in order to get all the backsoutnodes for a push (unless json-automationrelevance for the tip of a push gives us info about all commits of the push)

Reply:
Here's an example push with three backouts: https://treeherder.mozilla.org/#/jobs?repo=autoland&revision=c078d08afa05139cdd28659604214a3bf79adde9.

I don't see a way out of this, the push revision here is c078d08afa05139cdd28659604214a3bf79adde9, but there are also 1a1332949d44be803d9c065399b0883cadd5ff8c and 63acd82c9360389a6dbd3bdf223e48bf32ca25c1.

If we have a failure where 'fixed by commit' is 1a1332949d44be803d9c065399b0883cadd5ff8c, we can't build a push instance with 1a1332949d44be803d9c065399b0883cadd5ff8c as there is no push with this revision. We'd have to first perform a request to get the push containing 1a1332949d44be803d9c065399b0883cadd5ff8c, then create a Push instance with the data returned by hgmo. So in the end we'd have too many requests to hgmo.

Reply:
Good point. Could we keep this out of `push.py` at least then? Maybe `util/hgmo.py` or something like that?

Reply:
You no what, I have a patch in-progress that was moving the URL handling to a util file anyway, so just land this as is and I'll move it over later.

Reply:
Sounds good, and hopefully in the future we can just build a backout map right away using ActiveData (https://github.com/mozilla/mozci/issues/38).